integration-tests/applications/src/test/java/com/xsk/integration/tests/applications/deployment/XSKProjectDeployer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.xsk.integration.tests.applications.deployment;
+
+import com.xsk.integration.tests.applications.Deployment;
+import com.xsk.integration.tests.applications.deployment.client.PublisherClient;
+import com.xsk.integration.tests.applications.deployment.client.WorkspaceClient;
+import com.xsk.integration.tests.applications.deployment.client.XSKHttpClient;
+import org.apache.http.HttpResponse;
+
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static com.xsk.integration.tests.applications.Deployment.LOCAL;
+
+public class XSKProjectDeployer {
+
+    private final WorkspaceClient workspaceClient;
+    private final PublisherClient publisherClient;
+    private final Deployment xskDeployment;
+
+    public XSKProjectDeployer(Deployment deployment) {
+        XSKHttpClient xskHttpClient = new XSKHttpClient(deployment);
+        this.xskDeployment = deployment;
+        this.workspaceClient = new WorkspaceClient(xskHttpClient);
+        this.publisherClient = new PublisherClient(xskHttpClient);
+    }
+
+    public void deploy(String applicationName, Path applicationFolderPath) throws DeploymentException {
+        try {
+            deployAsync(applicationName, applicationFolderPath).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DeploymentException("Publish " + applicationName + " to " + applicationFolderPath + " failed", e);
+        }
+    }
+
+    public CompletableFuture<HttpResponse> deployAsync(String projectName, Path projectFolderPath) throws DeploymentException {
+        if (xskDeployment == LOCAL) {
+            workspaceClient.login();
```

Comment:
This future is not used in a `thenCompose` chain but it should as it's a dependency for other API calls. Also, couldn't we move this login functionality similarly to the Kyma token requests in the `XSKHttpClient`?